### DISCUSSION TITLE
Function in Node works but in Electron does not

### DISCUSSION THREAD

**Discussion by vikingBiking** [atom/atom#22389] [Q&A]
Hi!

I'm running the same function from both Electron and Node, but obtaining different results:

Invoking the function using Node from command line works fine, it lists the attached devices:

$> node my_func.js

```
var iosDevice = require("node-ios-device");

iosDevice.devices(function (err, devices) {
  if (err) {
    console.error("Error!", err);
  } else {
    console.log(devices);
  }
});
```

But putting the same chunk of code in main.js file in Electron yields the following outcome:

- The console does not show any result.
- Apparently, three instances of the app get created (Three app icons pop up in the dock).
<img width="173" alt="3_instances" src="https://user-images.githubusercontent.com/83988802/117735375-f1946600-b1ba-11eb-9b41-8f413f58b210.png">

this is the package.json

```
{
  "name": "test",
  "version": "1.0.0",
  "description": "",
  "main": "main.js",
  "scripts": {
    "start": "electron ."
  },
  "author": "Viking",
  "license": "ISC",
  "devDependencies": {
    "electron": "^11.4.4"
  },
  "dependencies": {
    "node-ios-device": "^1.9.2"
  }
```
}

Any idea about how I can accomplish the same task using Electron?

Thank you.

**Comment by TNGTJL** [comment]
Hello. Honestly I don’t know much about coding. I see things in colors so I
fall into it. I am learning though and typically I learn very fast but
coding is a lot of info so it’s quite challenging to dive into. I would
appreciate any kind of input on this particular post though
